sympy/sets/sets.py

```diff
@@ -674,7 +679,10 @@ def _measure(self):
 
     @property
     def _kind(self):
-        raise NotImplementedError("(%s)._kind" % self)
+        if all(i.kind == self.args[0].kind for i in self.args):
+            return SetKind(self.args[0].kind)
```

Comment:
Probably all cases like this will need to handle EmptySet:
```python
In [2]: Union({1}, EmptySet, evaluate=False)
Out[2]: {1} ∪ ∅
```
Maybe there should be a method defined on the Set class for getting the common element kind of a list of Sets so this code doesn't need to be duplicated everywhere.

Reply:
Current complilation give this
```python
>>> Union({1}, EmptySet).kind
SetKind(NumberKind)
>>> Union({1}, EmptySet, evaluate=False).kind
SetKind(UndefinedKind)
```
should `Union({1}, EmptySet, evaluate=False).kind` also give `SetKind(NumberKind)`

Reply:
@oscarbenjamin

Reply:
Yes, it should.